diff --git a/fluent-fallback/src/cache.rs b/fluent-fallback/src/cache.rs
index 32bc33f..1c47f66 100644
--- a/fluent-fallback/src/cache.rs
+++ b/fluent-fallback/src/cache.rs
@@ -89,10 +89,14 @@ where
             }
             Ordering::Equal => {
                 // Get the next item from the iterator
-                let item = self.cache.iter.borrow_mut().next();
-                self.curr += 1;
-                if let Some(item) = item {
-                    Some(self.cache.push_get(item))
+                if let Ok(mut iter) = self.cache.iter.try_borrow_mut() {
+                    let item = iter.next();
+                    self.curr += 1;
+                    if let Some(item) = item {
+                        Some(self.cache.push_get(item))
+                    } else {
+                        None
+                    }
                 } else {
                     None
                 }
